### DISCUSSION TITLE
Перестают резолвиться незаблокированные сайты.

### DISCUSSION THREAD

**Discussion by Selys-u** [xtrime-ru/antizapret-vpn-docker#103] [Q&A] [answered]
После запуска антизапрет + амнезия + адгуард через время перестают резолвится незаблокированные домены, например ya.ru, vk.com и т.д., при этом резолв заблокированных работает. Получается что все блокнутое открывается, а не блокнутое нет. 

```
С роутера Keenetic (ovpn)
_sending ICMP ECHO request to google.com...
Failed to resolve "google.com"._

_sending ICMP ECHO request to x.com...
PING x.com (10.224.0.72) 56 (84) bytes of data.
84 bytes from x.com (10.224.0.72): icmp_req=1, ttl=58, time=62.86 ms._
```

Если просто через AmneziaWG на телефоне, то тоже самое.

На самом сервере все резолвится и в самом докере антизапрета все резолвится.

Подскажите, куда копать?

**Comment by Selys-u** [comment]
После написания поста, минут через 20-30 сайты стали открываться, домены резолвятся... Пока наблюдаю

**Reply by Selys-u** [reply]
Через небольшой промежуток времени опять перестало резовлить незаблокированные сайты..(

**Reply by Selys-u** [reply]
tcpdump из контейнера с WG
```
10.1.166.3.56679 > dns.google.53: [udp sum ok] 41525+ A? vk.com. (24)
12:15:49.232283 wg0   Out IP (tos 0x0, ttl 63, id 22352, offset 0, flags [none], proto UDP (17), length 52)
    dns.google.53 > 10.1.166.3.56679: [bad udp cksum 0xc045 -> 0x0059!] 41525 ServFail q: A? vk.com. 0/0/0 (24)
12:15:49.333134 wg0   In  IP (tos 0x0, ttl 64, id 58834, offset 0, flags [none], proto UDP (17), length 52)
```

**Reply by Selys-u** [reply]
С указанием DNS сервера в ручную 
```
10.1.166.3.58523 > 10.224.0.1.53: [udp sum ok] 57586+ A? vk.com. (24)
12:17:47.649280 wg0   Out IP (tos 0x0, ttl 63, id 39619, offset 0, flags [none], proto UDP (17), length 52)
    10.224.0.1.53 > 10.1.166.3.58523: [bad udp cksum 0xbb16 -> 0xbf96!] 57586 ServFail q: A? vk.com. 0/0/0 (24)
12:17:50.538838 wg0   Out IP (tos 0x0, ttl 63, id 40027, offset 0, flags [none], proto UDP (17), length 52)
```

**Reply by RazGame** [reply]
Похоже, что такая проблема и с OVPN начала происходить

**Reply by Selys-u** [reply]
У меня вообще нигде не работает: не с OVPN ни с WG. 

Просто, время от времени, то резолвит, то не резолвит

**Comment by Murakumo-JP** [comment]
Вы нагрузку на vps сервер смотрели в этот момент ? У меня по непонятным причинам раз в пару суток начинается какая то ебанина процессор и оперативка на сервере забиваются под 100% я даже по ssh на сервер не могу попасть в этот момент только перезагрузка из консоли сайта спасает, ну и да в этот момент впн как раз помирает, думаю завтра снесу систему и поставлю все с нуля и буду наблюдать.

**Reply by Selys-u** [reply]
У меня нагрузки, в целом, нет. Но все равно тормоза в консоли по ssh. Но у меня, как я ниже выразился на хостера атаку проводят.

**Comment by crubik** [comment]
аналогично. openvpn дико тупит. что-то с днс. резволвы очень тупят
https://browserleaks.com/dns
постоянно меняются

по UDP не работает коннект, только по tcp

![Без имени](https://github.com/user-attachments/assets/34aa85d8-961e-43b9-a38c-df502df0089d)

**Comment by RazGame** [comment]
Может пора из дискуссии в issue переехать ?

**Comment by dartraiden** [comment]
Иногда перестаёт резолвить по какой-то причине.

Например, только что не удавалось отрезолвить `asio4all.org`, в то время как `linux.org.ru` резолвился. Но если зайти на сервер и из консоли отрезолвить (не в контейнере), то `asio4all.org` резолвился нормально.

Лечится это переподключением OVPN на роутере.

У antizapret.prostovpn.org, к слову, такая же проблема - пользователи жалуются, что иногда резолв отрубается полностью минут на 30 или больше, причём, похоже, это не зависит от клиента, т.к. жалуются и владельцы Keenetic, и владельцы Mikrotik. Главный подозреваемый пока - kresd.

**Reply by dartraiden** [reply]
Отпишусь: проблема полностью ушла, когда перешёл с OVPN на WG. Уже 2 месяца ни разу не сталкивался.

**Comment by Selys-u** [comment]
Товарищи, на моего хостера, к тому же оказалось, проводится масштабная DDoS атака, уже 2 дня. Вероятно, что связано с этим, но пока трудно сказать. Либо из-за этого, либо это совпадение, потому что нас не мало... Ну, либо у нас один хостер..

**Comment by Selys-u** [comment, marked as answer]
По всей видимости, нашел проблему у себя. Заключалась в том, что дефолтный Upstream DNS-сервер в настройках AdGuard Home периодически становится недоступным, точнее большую часть времени был недоступен (вероятно все из-за вышеописанной DDoS атаки на моего хостера, 1.1.1.1 тоже у меня не отвечают). Полез туда после того, как посмотрел логи контейнера, там было много error'ов. 

Заменил сервер на 8.8.8.8 и 8.8.4.4, пока полёт нормальный, уже около часа. Если все будет работать и дальше, то отмечу как решение на свой вопрос. 

Попробуйте у себя посмотреть, может тоже кому-то поможет.

**Reply by dartraiden** [reply]
Да, с какого-то момента контейнер почему-то использует 1.1.1.1 по умолчанию.
Раньше использовался DNS хостера.

**Reply by Selys-u** [reply]
Если у Вас нет AdGuard, то я так понимаю используется 1.1.1.1 по умолчанию, его можно сменить в docker-compose.yml, а если AdGuard есть, то тот адрес. который в настройках AdGuard. Выход - сменить DNS на рабочий

**Reply by RazGame** [reply]
хостинг pq?
мне помогло

**Reply by Selys-u** [reply]
Так точно, но периодически лагает. Потери большие из-за DDoS... (13% потерь)

**Reply by crubik** [reply]
> его можно сменить в docker-compose.yml

он там не используется
![Screenshot_2](https://github.com/user-attachments/assets/881156f0-d716-4c24-846f-06b334691d66)

**Reply by Selys-u** [reply]
> > его можно сменить в docker-compose.yml
> 
> он там не используется
> ![Screenshot_2](https://github.com/user-attachments/assets/881156f0-d716-4c24-846f-06b334691d66)
> 
> 

Да, все верно, дефолтно он и используется, как я понимаю. А если надо сменить, то надо раскоментировать и указать другие DNS

**Comment by dartraiden** [comment]
Вот, опять. Твич не резолвится ни на клиенте, ни на роутере, где поднят VPN. На сервере резолвится, в контейнере резолвится.

В сислоге на роутере
```
Окт 22 22:20:25
ndnproxy
[0AF6] answer from wrong socket (got from 34 195.208.4.1:53 [0] to 192.168.104.4:58369, exp from 15 192.168.104.1:53 to 56604).
Окт 22 22:20:25
ndnproxy
[0AF6] rogue answer from [195.208.4.1]:53 (expect from [192.168.104.1]:53).
Окт 22 22:23:29
ndnproxy
[0892] answer from wrong socket (got from 16 9.9.9.10:53 [0] to 192.168.104.4:54641, exp from 29 192.168.104.1:53 to 49436).
Окт 22 22:23:29
ndnproxy
[0892] rogue answer from [9.9.9.10]:53 (expect from [192.168.104.1]:53). 
```

195.208.4.1 и 9.9.9.10 это DNS. которые указаны на роутере и при поднятом VPN заворачиваются через директиву `route ...`

Прошло 10 минут и резолв ожил.

**Reply by Selys-u** [reply]
Поменяли DNS в AdGuard или в конфиге докера?

**Reply by dartraiden** [reply]
В конфиге докера у меня 8.8.8.8

**Reply by Selys-u** [reply]
А adguard home используете?

**Reply by dartraiden** [reply]
Нет. Просто OpenVPN и больше ничего

**Reply by Selys-u** [reply]
Только твитч? Странно, может у вас какие-то исключения или что-то подобное? В ином случае, надо у автора данного проекта уточнять, что может быть. 

А вы когда ДНС прописали контейнер пересоздали или просто ребутнули?

**Reply by dartraiden** [reply]
Нет, любые незаблокированные перестали разолвиться. У топикстартера та же проблема.

Да, пересоздал. Я после смены DNS на 8.8.8.8 случайно обнаружил, что докер заботливо сохранил предыдущие версии контейнеров (это мой первый опыт общения с докером), поэтому убил вообще все контейнеры и пересоздал.

**Reply by Selys-u** [reply]
Можно просто 
`docker-compose stop`
`docker-compose up -d --force-recreate`

**Comment by alexunderboots** [comment]
Похожая периодическая проблема, началась после обновления на актуальную, замечаю ее ночью:
2-hop, ovpn, резольвинг отрабатывает, трэйсе идёт, однако fwd сайты не открываются, curl висит в trying ..... иногда выскакивает сообщение об ошибке SSL (точно уже не вспомню), thunderbird в этот момент выдал ошибку на серт gmail указава домен googleapis.com

мониторинг проблему не замечает, тк icmp отрабатывает, добавлю проверку tls

Прописал DNS quad9, dc down&up - всё завелось, наблюдаю

**Reply by alexunderboots** [reply]
мониторинг регулярно ругается на сайты — TLS handshake timeout
Хотя нет, нехватало push routes на втором хопе

**Reply by alexunderboots** [reply]
www.google.com uses an unsupported protocol.
ERR_SSL_VERSION_OR_CIPHER_MISMATCH

**Reply by alexunderboots** [reply]
Фиксация DNS — помогла

**Reply by alexunderboots** [reply]
нет не помогла
`nslookup www.youtube.com 
Server:         10.90.254.254
Address:        10.90.254.254#53

Non-authoritative answer:
Name:   www.youtube.com
Address: 10.224.0.32
Name:   www.youtube.com
Address: 10.224.0.36
Name:   www.youtube.com
Address: 10.224.0.41
Name:   www.youtube.com
Address: 10.224.0.42
Name:   www.youtube.com
Address: 10.224.0.43
Name:   www.youtube.com
Address: 10.224.0.44
Name:   www.youtube.com
Address: 10.224.0.53
Name:   www.youtube.com
Address: 10.224.0.101
Name:   www.youtube.com
Address: 10.224.0.102
Name:   www.youtube.com
Address: 10.224.0.103
Name:   www.youtube.com
Address: 10.224.0.104
Name:   www.youtube.com
Address: 10.224.0.105
Name:   www.youtube.com
Address: 10.224.0.18
Name:   www.youtube.com
Address: 10.224.0.20
Name:   www.youtube.com
Address: 10.224.0.23
Name:   www.youtube.com
Address: 10.224.0.31`

`curl -v www.youtube.com/
* Host www.youtube.com:80 was resolved.
* IPv6: (none)
* IPv4: 10.224.0.32, 10.224.0.41, 10.224.0.103, 10.224.0.36, 10.224.0.44, 10.224.0.53, 10.224.0.101, 10.224.0.102, 10.224.0.42, 10.224.0.43, 10.224.0.104, 10.224.0.18, 10.224.0.20, 10.224.0.23, 10.224.0.31, 10.224.0.105
*   Trying 10.224.0.32:80...
* Connected to www.youtube.com (10.224.0.32) port 80
> GET / HTTP/1.1
> Host: www.youtube.com
> User-Agent: curl/8.7.1
> Accept: */*
> 
* Request completely sent off
< HTTP/1.1 402 Payment Required
< Server: nginx
< Date: Wed, 08 Jan 2025 15:17:53 GMT
< Content-Type: text/html
< Content-Length: 160
< Connection: keep-alive
< 
<html>
<head><title>402 Payment Required</title></head>
<body>
<center><h1>402 Payment Required</h1></center>
<hr><center>nginx</center>
</body>
</html>`

down&up не помог, минут через 30 отпустило

**Reply by alexunderboots** [reply]
перевел на WG, через пару часов хостинг упал, после загрузки ошибка в журнале
kernel: watchdog: BUG: soft lockup - CPU#0 stuck for 27s! [kresd:2007]

Между падением и восстановлением 20 минут, в журнале записей нет
